source/databricks/calculation_engine/tests/calculation/energy/test_transform_quarter_to_hour.py

```diff
@@ -0,0 +1,201 @@
+    observation_time: datetime = datetime(2020, 1, 1, 0, 0),
+    quantity: Decimal = Decimal("4.444000"),
+    quality: QuantityQuality = QuantityQuality.ESTIMATED,
+    energy_supplier_id: str = "the_energy_supplier_id",
+    balance_responsible_id: str = "the_balance_responsible_id",
+    settlement_method: str = "the_settlement_method",
+) -> Row:
+    row = {
+        Colname.grid_area: grid_area,
+        Colname.to_grid_area: to_grid_area,
+        Colname.from_grid_area: from_grid_area,
+        Colname.metering_point_id: metering_point_id,
+        Colname.metering_point_type: metering_point_type,
+        Colname.resolution: resolution.value,
+        Colname.observation_time: observation_time,
+        Colname.quantity: quantity,
+        Colname.quality: quality.value,
+        Colname.energy_supplier_id: energy_supplier_id,
+        Colname.balance_responsible_id: balance_responsible_id,
+        Colname.settlement_method: settlement_method,
+    }
+
+    return Row(**row)
+
+
+def test__transform_quarter_to_hour__when_valid_input__merge_basis_data_time_series(
+    spark: SparkSession,
+) -> None:
+    # Arrange
+    rows = [
+        basis_data_time_series_points_row(resolution=MeteringPointResolution.HOUR),
+        basis_data_time_series_points_row(resolution=MeteringPointResolution.QUARTER),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 0, 15),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 0, 30),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 0, 45),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 1, 00),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 1, 15),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 1, 30),
+        ),
+        basis_data_time_series_points_row(
+            resolution=MeteringPointResolution.QUARTER,
+            observation_time=datetime(2020, 1, 1, 1, 45),
+        ),
+    ]
+    basis_data_time_series_points = spark.createDataFrame(
+        rows, prepared_metering_point_time_series_schema
+    )
+    metering_point_time_series = PreparedMeteringPointTimeSeries(
+        basis_data_time_series_points
+    )
+
+    # Act
+    actual = transform_quarter_to_hour(metering_point_time_series)
+
+    # Assert
+    assert actual.df.count() == 3
+    assert actual.df.collect()[0][Colname.quantity] == Decimal("4.444000")
```

Comment:
If you move the factory to separate module and provide the defaults, this could become something like:
```suggestion
    assert actual.df.collect()[0][Colname.quantity] == factory.DEFAULT_QUANTITY
```

Reply:
The idea applies to the other asserts as well.

Reply:
i cant do that since the prepared metering point time series factory is using another default, but i made a default in the top of the file and is using that instead